File: src/backend/Backend.Domain/Veiculos/Entities/Veiculo.cs
using Backend.Domain.Bases.Entities;

namespace Backend.Domain.Veiculos.Entities;

public class Veiculo : BaseEntity
{
    public string Nome { get; set; }
    public decimal Valor { get; set; }
    public string Imagem { get; set; }
    public Guid? MarcaId { get; set; }
    public Guid? ModeloId { get; set; }

    /* EF Relations */
    public Marca? Marca { get; set; }
    public Modelo? Modelo { get; set; }
}

File: src/frontend/src/app/veiculos/models/veiculo.ts
import { Marca } from "./marca";
import { Modelo } from "./modelo";

export interface Veiculo{
  id: string;
  nome: string;
  valor: number;
  marcaId: string;
  modeloId: string;
  uploadDaImagem: string;
  nomeDaImagem: string,

  marca: Marca;
  modelo: Modelo;
}
